internal/appconfig/watcher.go

```diff
@@ -852,6 +852,40 @@ func buildAppConfig(
 					}
 				}
 
+			case *sdkpb.ConfigSource_Value_Json:
+				if req.cv.Internal {
+					// We don't yet support using non-string types as internal vals.
+					// Logging and moving along isn't great though - the user
+					// won't have much indication as to why their variable isn't working.
+					L.Warn("Complex-typed outputs cannot be used as internal values. Skipping.", "var name", req.cv.Name)
+				} else {
```

Comment:
I think we should probably in-line the remaining if-statements to be `else if`? I don't think we need this else at least

Reply:
Updated for clarity 👍